android/src/main/kotlin/com/onegini/mobile/sdk/flutter/providers/CustomTwoStepRegistrationActionImpl.kt

```diff
@@ -0,0 +1,45 @@
+package com.onegini.mobile.sdk.flutter.providers
+
+import com.google.gson.Gson
+import com.onegini.mobile.sdk.android.handlers.action.OneginiCustomRegistrationAction
+import com.onegini.mobile.sdk.android.handlers.action.OneginiCustomTwoStepRegistrationAction
+import com.onegini.mobile.sdk.android.handlers.request.callback.OneginiCustomRegistrationCallback
+import com.onegini.mobile.sdk.android.model.entity.CustomInfo
+import com.onegini.mobile.sdk.flutter.constants.Constants
+import com.onegini.mobile.sdk.flutter.helpers.OneginiEventsSender
+import com.onegini.mobile.sdk.flutter.models.CustomRegistrationModel
+import com.onegini.mobile.sdk.flutter.models.OneginiEvent
+
+class CustomTwoStepRegistrationActionImpl(private val providerId: String) : OneginiCustomTwoStepRegistrationAction, CustomRegistrationAction {
+    var callback: OneginiCustomRegistrationCallback? = null
+
+    override fun initRegistration(callback: OneginiCustomRegistrationCallback, info: CustomInfo?) {
+        this.callback = callback
+
+        val data = Gson().toJson(CustomRegistrationModel(info?.data.orEmpty(), info?.status, providerId))
+        OneginiEventsSender.events?.success(Gson().toJson(OneginiEvent(Constants.EVENT_INIT_CUSTOM_REGISTRATION, data)))
+    }
+
+    override fun finishRegistration(callback: OneginiCustomRegistrationCallback, customInfo: CustomInfo?) {
+        this.callback = callback
+
+        val data = Gson().toJson(CustomRegistrationModel(customInfo?.data.orEmpty(), customInfo?.status, providerId))
+        OneginiEventsSender.events?.success(Gson().toJson(OneginiEvent(Constants.EVENT_FINISH_CUSTOM_REGISTRATION, data)))
+    }
+
+    override fun getCustomRegistrationAction(): OneginiCustomRegistrationAction {
+        return this
+    }
+
+    override fun getIdProvider(): String {
+        return providerId
+    }
+
+    override fun returnSuccess(result: String?) {
+        this.callback?.returnSuccess(result)
```

Comment:
callback should be set to null after this function

Reply:
done